client/packages/common/src/intl/utils/DateUtils.ts

```diff
@@ -60,6 +60,18 @@ const formatIfValid = (
   }
 ): string => (isValid(date) ? format(date, dateFormat, options) : '');
 
+// Adds the current time to a date object (that presumably has 00:00 as its time
+// component) -- does not mutate input Date object)
+const addCurrentTime = (date: Date | null): Date | null => {
+  if (date === null) return date;
+  const d = new Date();
+  const e = new Date(d);
+  const msSinceMidnight = d.getTime() - e.setHours(0, 0, 0, 0);
```

Comment:
See comment below. I would avoid confusion with e here be inlining it:
```suggestion
  const msSinceMidnight = d.getTime() - new Date(d).setHours(0, 0, 0, 0);
```

Reply:
7c46dfb59

Reply:
58592866a